Report_Generation/Report_graphs.py

```diff
@@ -40,37 +48,116 @@ def query_data(self, start_date, end_date, ticker,user,password,host,port,databa
+        self.monthly()
+        self.hourly()
+        print('Complete')
+        self.output.close()
+
+    def date_sep(self,data):
+        temp = (data['Date'].str.split('-').tolist())
+        return ['/'.join(i[1:]) for i in temp]
+
+    def price_change(self):
+        plt.plot(self.date,self.data['change'])
+        plt.text(0.05,0.95,'This Plot shows the changes in price with relation to the previous' +\
+            ' day',transform=self.fig.transFigure)
+        plt.xlabel('Date')
+        plt.ylabel('Price($) change with relation to previous day')
+        plt.xticks(fontsize = 8, rotation = 90)
+        self.output.savefig()
+        plt.clf()
+        return 
+
+    def date_graph(self):
+        temp_data = self.data.groupby('Day', as_index = False ).mean()
+        # plt.xticks(fontsize=8, rotation=90)
+        plt.bar(temp_data['Day'],temp_data['change'], align = 'center')
+        plt.xticks(range(5),['Monday','Tuesday','Wednesday','Thursday','Friday'],rotation = 'vertical')
+        plt.title('Trend in of change of stock price with day of week')
+        plt.xlabel('Day of the week')
+        plt.ylabel('Average Price Change($)')
+        self.output.savefig()
+        plt.clf()
+        return 
+
+    def sep_time(self,data,separator,param,new_name,return_index):
+        temp  = (data[param].str.split(separator).tolist())
+        data[new_name] = [value[return_index] for value in temp]
+        return data
 
     def hourly(self):
-        pass
+        new_data = self.sep_time(self.data,':','Time','hour',0)
+        new_data = new_data.groupby('hour', as_index = False).mean()
+        plt.title('Average Hourly Price Change')
+        plt.xticks(fontsize=8, rotation=90)
+        txt = 'The average price change for each hour between ' + self.start + ' and ' + self.end 
+        plt.xlabel('Date')
+        plt.ylabel('Average Hourly Price($)')
+        plt.text(0.1, 0.95, txt, transform = self.fig.transFigure)
+        plt.bar(new_data['hour'],new_data['change'], align = 'center')
+        plt.xticks(range(len(new_data['hour'])),[i for i in new_data['hour']],rotation = 'vertical')
+        self.output.savefig()
+        plt.clf()
+        return 
 
     def monthly(self):
-        pass
+        new_data = self.sep_time(self.data,'-','Date','month',1)
+        new_data = new_data.groupby('month',as_index = False).mean()
+        plt.title('Average Monthly price Increment')
+        plt.xticks(fontsize=8, rotation=90)
+        txt = 'The average price change per month between ' + self.start + ' and ' + self.end 
+        plt.xlabel('Date')
+        plt.bar(new_data['month'],new_data['change'], align = 'center')
+        plt.xticks(range(len(new_data['month'])),[i for i in new_data['month']],rotation = 'vertical')
+        plt.ylabel('Average Monthly Price($)')
+        plt.text(0.1, 0.95, txt, transform = self.fig.transFigure)
+        self.output.savefig()
+        plt.clf()
+        return
 
     def weekday(self):
```

Comment:
looks to me like you can extract the monthly, hourly, and weekday plot steps into a common function